conans/test/functional/toolchains/microsoft/test_msbuilddeps.py

```diff
@@ -396,6 +398,12 @@
 """
 
 
+vs_versions = [{"vs_version": "15", "msvc_version": "19.1", "ide_year": "2017", "toolset": "v141"}]
```

Comment:
The functional tests for XXXDeps do not need to be repeated for every compiler version, they are practically independent of the VS version, and they are very slow. The way is to parameterize the test with the default VS version, and run only once. If anything some fast unit-test, that doesn't build anything.

Reply:
they seem to be dependent on VS version to some degree: e.g. they pass `-s compiler.version` at very least, also, it seems like they invoke the actual build with the actual compiler. 
(I was expecting generator tests to just generate some files and validate their contents, but these tests seems to be going to verify the actual use case of consuming generating output and trying to use it with the real solution).
more, the toolchain right now tries to evaluate vcvars (`_write_conanvcvars`), so it wants a valid mapping of toolset -> vcvars version, and it actually even tries to compute `vs_installation_path` (from `vcvars_path`) - yes, debugger hits that from the tests.
so practically speaking, these tests actually depend on the different VS versions.
and yes, they are very slow, but CI skips vs2022 tests anyway, so right now it's not an issue IMO.